ting-bill-split.go

```diff
@@ -362,35 +364,166 @@ func createBillsFile(path string) {
 	}
 }
 
-func main() {
-	fmt.Printf("Ting Bill Splitter\n\n")
+func isFileMatch(fileName string, nameTerm string, ext string) bool {
+	r := regexp.MustCompile(`(?i)^[\w-]*` + nameTerm + `[\w-]*$`)
+
+	if ext != "" {
+		r = regexp.MustCompile(`(?i)^[\w-]*` + nameTerm + `[\w-]*(\.` + ext + `)$`)
+	}
+
+	return r.MatchString(fileName)
+}
+
+func parseDir(path string) {
+	files, err := ioutil.ReadDir(path)
+
+	var billFile *os.File
+	var minFile *os.File
+	var msgFile *os.File
+	var megFile *os.File
 
-	billPtr := flag.String("bills", "", "filename for bills toml - ex: -bills=\"bills.toml\"")
-	minPtr := flag.String("minutes", "", "filename for minutes csv - ex: -minutes=\"minutes.csv\"")
-	msgPtr := flag.String("messages", "", "filename for messages csv - ex: -messages=\"messages.csv\"")
-	megPtr := flag.String("megabytes", "", "filename for megabytes csv - ex: -megabytes=\"megabytes.csv\"")
+	if err != nil {
+		log.Fatal(err)
+	}
 
-	badParam := false
-	paramMap := map[string]*string{
-		"bills":     billPtr,
-		"minutes":   minPtr,
-		"messages":  msgPtr,
-		"megabytes": megPtr,
+	for _, file := range files {
+		if !file.IsDir() {
+			fmt.Println(file.Name())
+
+			if billFile == nil && isFileMatch(file.Name(), "bills", "toml") {
+				billFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(billFile)
+			}
+
+			if minFile == nil && isFileMatch(file.Name(), "minutes", "csv") {
+				minFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(minFile)
+			}
+
+			if msgFile == nil && isFileMatch(file.Name(), "messages", "csv") {
+				msgFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(msgFile)
+			}
+
+			if megFile == nil && isFileMatch(file.Name(), "megabytes", "csv") {
+				megFile, err = os.Open(file.Name())
+				if err != nil {
+					log.Fatal(err)
+				}
+				fmt.Println(megFile)
+			}
+		}
 	}
 
+	if billFile == nil || minFile == nil || msgFile == nil || megFile == nil {
+		fmt.Println("Unable to open necessary files.")
+
+		if billFile == nil {
+			fmt.Println("Bills file not found.")
+		}
+
+		if minFile == nil {
+			fmt.Println("Minutes file not found.")
+		}
+
+		if msgFile == nil {
+			fmt.Println("Messages file not found.")
+		}
+
+		if megFile == nil {
+			fmt.Println("Megabytes file not found.")
+		}
+	} else {
```

Comment:
Don't use `else`. Make the previous block `return` if any of the files were missing.